hbase-common/src/main/java/org/apache/hadoop/hbase/io/crypto/tls/X509Util.java

```diff
@@ -131,6 +137,34 @@ private static String[] getCBCCiphers() {
   private static final String[] DEFAULT_CIPHERS_JAVA9 =
     ObjectArrays.concat(getGCMCiphers(), getCBCCiphers(), String.class);
 
+  private static final String[] DEFAULT_CIPHERS_OPENSSL = getOpenSslFilteredDefaultCiphers();
+
+  /**
+   * Not all of our default ciphers are available in OpenSSL. Takes our default cipher lists and
+   * filters them to only those available in OpenSsl. Does GCM first, then CBC because GCM tends to
+   * be better and faster, and we don't need to worry about the java8 vs 9 performance issue if
+   * OpenSSL is handling it.
+   */
```

Comment:
Nice approach. In ZooKeeper we simply don't set default cipher suites just let OpenSSL to decide. User can still override them though.

Reply:
Yea I went with this approach because I'm really not an SSL expert and dont have any opinions about the cipher suites used here or what's in openssl as defaults. So I figured I could make a least-invasive change which uses what was defaulted here an just filters to those that work.